File: Arrays/2965-Find-Missing-and-Repeated-Value/JavaScript/2965-Find-Missing-and-Repeated-Values-Brute Force.js
/**
 * Problem: 2965. Find Missing and Repeated Values
 * Approach: Brute Force
 *
 * Time Complexity: O(n^4)
 * Space Complexity: O(1)
 */

/**
 * @param {number[][]} grid
 * @return {number[]}
 */

var findMissingAndRepeatedValues = function(grid) {

    let n = grid.length;

    let repeating = -1;
    let missing = -1;

    for (let num = 1; num < n * n + 1; num++) {

        let count = 0;

        for (let i = 0; i < n; i++) {
            for (let j = 0; j < n; j++) {

                if (grid[i][j] === num) {
                    count++;
                }
            }
        }

        if (count === 2) {
            repeating = num;
        }
        else if (count === 0) {
            missing = num;
        }

        if (repeating !== -1 && missing !== -1) {
            break;
        }
    }

    return [repeating, missing];
};

File: Arrays/2965-Find-Missing-and-Repeated-Value/Python/2965-Find-Missing-and-Repeated-Values-Brute Force.py
"""
Problem: 2965. Find Missing and Repeated Values
Approach: Brute Force

Time Complexity: O(n^4)
Space Complexity: O(1)
"""

class Solution:
    def findMissingAndRepeatedValues(self, grid: list[list[int]]) -> list[int]:

        n = len(grid)

        repeating = -1
        missing = -1

        for num in range(1, n * n + 1):

            count = 0

            for row in grid:
                for value in row:
                    if value == num:
                        count += 1

            if count == 2:
                repeating = num

            elif count == 0:
                missing = num

            if repeating != -1 and missing != -1:
                break

        return [repeating, missing]